### PR TITLE
move doNotDedup to experimental

### PR DESCRIPTION
**Pull request by sequencer**
**Related issue**: https://github.com/freechipsproject/firrtl/issues/998

**Type of change**: other enhancement

**Impact**: API addition (no impact on existing code)

**Development Phase**: proposal

**Release Notes**
`doNotDedup` has be implemented in ` firrtl/src/main/scala/firrtl/transforms/Dedup.scala`, and tested in `chisel3/src/test/scala/chiselTests/AnnotationNoDedup.scala`. This commit move it to experimental as an API.
and maybe we need to fit the `dontTouch` name, rename this API to `dontDedup`?

### CODE REVIEW

**Comment by ucbjrl** [comment]
ok to test

**Comment by sequencer** [comment]
ping

**Comment by edwardcwang** [comment]
Slated for discussion soon, ideally for next week

**Comment by sequencer** [comment]
Thx

File: chiselFrontend/src/main/scala/chisel3/core/Annotation.scala [hunk 1]
```diff
@@ -88,3 +89,49 @@ object dontTouch { // scalastyle:ignore object.name
...
+  *  val io = IO(new Bundle {
+  *    val a = Input(UInt(32.W))
+  *    val b = Input(UInt(32.W))
+  *    val out = Output(Vec(2, UInt(32.W)))
+  *  })
+  *
+  *  io.out(0) := fullAdder(io.a, io.b, "mod1")
+  *  io.out(1) := fullAdder(io.a, io.b, "mod2")
+  *}
+  *
```
Comment (jackkoenig) [inline chiselFrontend/src/main/scala/chisel3/core/Annotation.scala]: Needs closing `}}}` for ScalaDoc I think

**Comment by jackkoenig** [comment]
Thanks :)

**Pull request merged**